### PR TITLE
Pull hollow node images from real nodes

### PR DESCRIPTION
**Pull request by lyzs90**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Details: https://github.com/kubernetes/kubernetes/issues/90833

**Which issue(s) this PR fixes**:

Part of https://github.com/kubernetes/kubernetes/issues/90833

**Special notes for your reviewer**:
- Was considering running the hollow node on a local cluster and get it to load some images. But sadly i'm on a mac and not sure if using dockershim will be a good enough test
- Otherwise would we be able to go straight for a presubmit if this looks good?

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by mm4tt** [comment]
Nice! Let's see whether it works in the kubemark presubmit.

File: cmd/kubemark/hollow-node.go [hunk 1]
```diff
@@ -222,12 +222,17 @@ func run(config *hollowNodeConfig) {
...
+			klog.Fatalf("Failed to init image service %v.", err)
+		}
+
 		hollowKubelet := kubemark.NewHollowKubelet(
 			f, c,
 			client,
 			heartbeatClient,
 			cadvisorInterface,
-			fakeRemoteRuntime.ImageService,
+			remoteImageService,
```
Comment (mm4tt) [inline cmd/kubemark/hollow-node.go:235]: Wouldn't this have an unwanted side effect that hollow nodes now will actually run the images scheduled on them?

**Comment by mm4tt** [comment]
After thinking about it I think it will fail as the kubemark jobs haven't been migrated to containerd :( 

Let's wait for the presubmit results to confirm. If so, I'll file an issue to migrate the kubemark jobs to containerd.

File: cmd/kubemark/hollow-node.go (hunk 1 above)
Comment (mm4tt) [reply cmd/kubemark/hollow-node.go:235]: Probably not, as there are separate interfaces for imageService and runtimeService... :)

**Comment by mm4tt** [comment]
The kubemark job will fail due to setup issues - https://github.com/kubernetes/kubernetes/pull/97858

File: cmd/kubemark/hollow-node.go (hunk 1 above)
Comment (lyzs90) [reply cmd/kubemark/hollow-node.go:235]: Yep runtime is still fake

**Comment by mm4tt** [comment]
The setup issue should be fixed now. Let's retry the test

/test pull-kubernetes-kubemark-e2e-gce-big

**Comment by mm4tt** [comment]
Ok, the test didn't fail, but that doesn't mean that it worked. Could you patch the https://github.com/kubernetes/kubernetes/pull/98098 to see what images are in the hollow nodes?

**Comment by lyzs90** [comment]
> Ok, the test didn't fail, but that doesn't mean that it worked. Could you patch the #98098 to see what images are in the hollow nodes?

great! ok will do so

**Comment by lyzs90** [comment]
@mm4tt i can't raise a PR onto your fork. should i pull your changes here instead?

**Comment by mm4tt** [comment]
> Matt Matejczyk i can't raise a PR onto your fork. should i pull your changes here instead?

Yeah, just make the same changes here as I did in my PR. These are just two lines

**Comment by lyzs90** [comment]
> > Matt Matejczyk i can't raise a PR onto your fork. should i pull your changes here instead?
> 
> Yeah, just make the same changes here as I did in my PR. These are just two lines

Done, you can refire the presubmit

**Comment by mm4tt** [comment]
Ha, it works!

[Baseline](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/98098/pull-kubernetes-kubemark-e2e-gce-big/1350066641148317696/):
```
I0115 13:34:39.359] hollow-node-22h4n                 [k8s.gcr.io/prometheus-to-sd:v0.5.0],[k8s.gcr.io/metadata-proxy:v0.1.12]
```

[This PR](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/97858/pull-kubernetes-kubemark-e2e-gce-big/1351109519295385600/):
```
I0118 10:38:28.105] hollow-node-229dq                 [k8s.gcr.io/kube-proxy-amd64:v1.21.0-alpha.1.117_4110ce3a8c0d7d],[gcr.io/k8s-infra-e2e-boskos-scale-16/kubemark@sha256:3457e249abd3690d07c201c7e481d736e9075d0e2617d5e350d578f04385da28 gcr.io/k8s-infra-e2e-boskos-scale-16/kubemark:oaewoa],[k8s.gcr.io/prometheus-to-sd@sha256:14666989f40bb7c896c3e775a93c6873e2b791d65bc65579f58a078b7f9a764e k8s.gcr.io/prometheus-to-sd:v0.5.0],[k8s.gcr.io/metadata-proxy@sha256:e914645f22e946bce5165737e1b244e0a296ad1f0f81a9531adc57af2780978a k8s.gcr.io/metadata-proxy:v0.1.12],[docker.io/library/busybox@sha256:c5439d7db88ab5423999530349d327b04279ad3161d7596d2126dfb5b02bfd1f docker.io/library/busybox:1.32],[k8s.gcr.io/pause@sha256:927d98197ec1141a368550822d18fa1c60bdae27b78b0c004f705f548c07814f k8s.gcr.io/pause:3.2]
```

Nice! In this case, let's revert the changes in start-kubemark.sh and submit this. In the meantime, I'll investigate how it works - I thought kubemark jobs are still based on docker.

**Comment by mm4tt** [comment]
Ok, I was wrong to say that kubemark jobs use docker. The default in kube-up has been changed to containerd in Jun 2020 - https://github.com/kubernetes/kubernetes/pull/91684

**Comment by lyzs90** [comment]
awesome, i didn't get why the last test run failed though

**Pull request merged**